lib/galaxy/tools/actions/__init__.py

```diff
@@ -236,7 +236,7 @@ def execute(self, tool, trans, incoming={}, return_job=False, set_output_hid=Tru
                 incoming[ "%s|__identifier__" % name ] = identifier
 
             for tag in data.tags:
-                if tag.user_tname == 'name':
+                if tag.user_tname == 'name' and tag.value not in [t.value for t in preserved_tags]:
```

Comment:
Yeah, there's this second path of logic for this other type of collection output, patch inc in a sec.